Guide/src/reference/architecture/openhcl_servicing.md

```diff
@@ -0,0 +1,258 @@
+# OpenHCL Servicing
+
+## Overview
+
+OpenHCL servicing (also known as VTL2 servicing) is a mechanism that allows the host to update or service the OpenHCL paravisor while minimizing downtime and maintaining device state. During a servicing operation, OpenHCL can save its state, shut down, be updated by the host, and then restore its previous state when restarted.
+
+This capability is particularly important in production environments where maintaining continuous VM operation during paravisor updates is critical.
+
+## Servicing Lifecycle
+
+A servicing operation follows these steps:
+
+### 1. Servicing Request
+
+The host initiates a servicing operation by sending a request to OpenHCL with:
+- **Correlation ID**: For tracing and diagnostics
+- **Timeout Hint**: Deadline for completing the save operation
+- **Capabilities Flags**: Indicates what the host supports
+
+### 2. State Saving
+
+OpenHCL pauses VM execution and saves state from multiple components:
+
+#### State Units
+- **VmBus Relay**: VMBus channel state and connections
+- **Device Workers**: Individual device driver states
+- **Chipset Devices**: Hardware emulation state
+- **Firmware State**: UEFI/BIOS runtime state
+
+#### Servicing Init State
+- **Firmware Type**: How the VM booted (UEFI/PCAT/None)
+- **VM Stop Reference Time**: Hypervisor reference time when state units stopped
+- **Emuplat State**: RTC, PCI bridge, and network VF manager state
+- **VMGS State**: Virtual machine guest state storage
+- **Correlation ID**: For tracing across the servicing operation
+
+#### Device-Specific State
+- **NVMe State**: NVMe manager and driver state (when keepalive is enabled)
+- **DMA Manager State**: Private and shared pool allocations
+- **VMBus Client State**: Client-side VMBus connection state
+- **MANA State**: Microsoft Azure Network Adapter state
+
+### 3. Component Shutdown
+
+After state is saved, components are shut down:
+
+#### Without Keepalive
+- NVMe devices are cleanly shut down
+- Device handles are closed
+- DMA allocations are released
+- VFIO device handles are dropped (causing device reset)
+
+#### With Keepalive (NVMe)
+- NVMe devices remain connected (CC.EN=1)
+- VFIO device handles are kept open (preventing reset)
+- DMA buffers in the private pool are preserved
+- Device maintains its operational state
```

Comment:
The level of detail here is incorrect. The sub-bullets (VmBus relay, device workers, etc.) should be considered just examples.